waku/incentivization/txid_proof.nim

```diff
@@ -0,0 +1,76 @@
+import std/options, chronos, web3, stew/byteutils, stint, results, chronicles
+
+import waku/incentivization/rpc
+
+const SimpleTransferGasUsed = Quantity(21000)
+
+proc eth_getTransactionByHash_async(
+    txHash: TxHash, web3: Web3
+): Future[TransactionObject] {.async.} =
+  await web3.provider.eth_getTransactionByHash(txHash)
+
+proc getMinedTransactionReceipt_async(
```

Comment:
In these two proc names, we don't really tend to add async to the name nor to use snake case.

Maybe we can name them `getTransactionByHash` and `getMinedTransactionReceipt`?